numpy/lib/histograms.py

```diff
@@ -1047,7 +1047,10 @@ def histogramdd(sample, bins=10, range=None, normed=None, weights=None,
                 raise ValueError(
                     '`bins[{}]` must be positive, when an integer'.format(i))
             smin, smax = _get_outer_edges(sample[:,i], range[i])
-            edges[i] = np.linspace(smin, smax, bins[i] + 1)
+            if type(bins[i]) != int:
```

Comment:
This is worse, because it now rejects np.int8 and similar. The only "right" way to check if something is a suitable integer is to use `operator.index`.